### ISSUE TITLE
Error compiling elmer on Fedora 27

### ISSUE THREAD

**Issue by Enricc** [ElmerCSC/elmerfem#127]
Hi, I get this error when compiling Elmer on Fedora 27 which uses gcc7. Should I install an older version of gcc?

/elmerfem/fem/src/H1Basis.F90:41:0:

   END SUBROUTINE H1Basis_GetEdgeDirection
 
internal compiler error: in fold_convert_loc, at fold-const.c:2269

Thanks

**Comment by mbycklin** [comment]
Seems like a gfortran compiler bug. Please report it in https://gcc.gnu.org/bugzilla/. 

Was OpenMP enabled? gcc7 only partially supports OpenMP 4.5 features which the H1Basis -module uses.  As a workaround, you can try using a different version of the compiler or disabling OpenMP during compilation.

**Comment by sjneph** [comment]
If you need OpenMP, this might get you by until gfortran fully supports OpenMP 4.5.  Change H1Basis.F90 at lines 50-54:
```
!#if _OPENMP>=201511
!#define LINEAR_REF(var) LINEAR(REF(var))
!#else
#define LINEAR_REF(var)
!#endif
```

**Comment by mbycklin** [comment]
By the OpenMP standard, _OPENMP should describe the OpenMP feature set the compiler is compliant with. Maybe the best way to fix the issue is to have CMake test for the existence of the OpenMP SIMD features used in the Elmer code and then set up the build accordingly.

**Comment by mbycklin** [comment]
The compiler issue was reported and should be fixed in the next version of GNU Fortran. For details, see https://gcc.gnu.org/bugzilla/show_bug.cgi?id=84418.

**Comment by mbycklin** [comment]
The compilation issue should be worked around in pull request #135. Please report back if you still have problems with the OpenMP SIMD constructs used.

**Comment by mbycklin** [comment]
Merged #135 to trunk, closing.
